webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModalSelectors.js

```diff
@@ -1,7 +1,11 @@
+import Immutable from 'seamless-immutable';
+
 export const selectForemanModalsState = state => state.foremanModals;
 export const selectModalStateById = (state, id) =>
-  state.foremanModals && state.foremanModals[id];
-export const selectIsModalOpen = (state, id) => {
-  const openState = selectModalStateById(state, id);
-  return openState && openState.open;
-};
+  state.foremanModals && (state.foremanModals[id] || Immutable({}));
+export const selectIsModalOpen = (state, id) =>
+  selectModalStateById(state, id).isOpen;
+export const selectIsModalSubmitting = (state, id) =>
+  selectModalStateById(state, id).isSubmitting;
+export const modalExists = (state, id) =>
```

Comment:
The selector should be named `selectModalExists` just to conform to the convention.  For readability elsewhere I'm fine with the extra line of code to assign its return value to a variable: `const modalExists = selectModalExists(...)`